Source/FieldSolver/ElectrostaticSolver.H

```diff
@@ -32,31 +41,81 @@ struct PhiCalculatorEB {
     }
 };
 
-class BoundaryValueHandler {
+class BoundaryHandler {
 
 public:
+
+    amrex::Array<amrex::LinOpBCType,AMREX_SPACEDIM> lobc, hibc;
+    bool bcs_set = false;
+    std::array<bool,AMREX_SPACEDIM> dirichlet_flag;
+    bool has_Dirichlet = false;
+    bool phi_EB_only_t = true;
+
+    // set default potentials to zero in order for current tests to pass
+    // but forcing the user to specify a potential might be better
+    std::string potential_xlo_str = "0";
+    std::string potential_xhi_str = "0";
+    std::string potential_ylo_str = "0";
+    std::string potential_yhi_str = "0";
+    std::string potential_zlo_str = "0";
+    std::string potential_zhi_str = "0";
     std::string potential_eb_str = "0";
-    // TODO add handling of domain boundaries as well
+
+    void definePhiBCs ();
 
     void buildParsers ()
     {
+        potential_xlo_parser = makeParser(potential_xlo_str, {"t"});
```

Comment:
Dare I ask, but can the potentials on the boundaries also be made spatially dependent? This is a capability I've used in Warp. Though, maybe that should be a separate PR.

Reply:
This can be done yes. It would require modifying the `setPhiBC` function in `ElectrostaticSolver.cpp` to properly set the boundary potential. I agree that such a change should probably go in a separate PR.